### PR TITLE
python3Packages.sanic: 21.12.1 -> 22.3.2

### PR DESCRIPTION
**Pull request by mweinelt**
###### Description of changes

and some related dependencies. Fixes Python 3.10 support.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Waiting for #174718

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 174716` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.pytest-sanic</li>
    <li>python39Packages.pytest-sanic</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.elastic-apm</li>
    <li>python310Packages.sanic-auth</li>
    <li>python39Packages.elastic-apm</li>
    <li>python39Packages.sanic-auth</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>dyndnsc</li>
    <li>python310Packages.entrance</li>
    <li>python310Packages.entrance-with-router-features</li>
    <li>python310Packages.json-logging</li>
    <li>python310Packages.sanic</li>
    <li>python310Packages.sanic-routing</li>
    <li>python310Packages.sanic-testing</li>
    <li>python39Packages.entrance</li>
    <li>python39Packages.entrance-with-router-features</li>
    <li>python39Packages.json-logging</li>
    <li>python39Packages.sanic</li>
    <li>python39Packages.sanic-routing</li>
    <li>python39Packages.sanic-testing</li>
  </ul>
</details>

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 174716` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.pytest-sanic</li>
    <li>python39Packages.pytest-sanic</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>dyndnsc</li>
    <li>python310Packages.elastic-apm</li>
    <li>python310Packages.entrance</li>
    <li>python310Packages.entrance-with-router-features</li>
    <li>python310Packages.json-logging</li>
    <li>python310Packages.sanic</li>
    <li>python310Packages.sanic-auth</li>
    <li>python310Packages.sanic-routing</li>
    <li>python310Packages.sanic-testing</li>
    <li>python39Packages.elastic-apm</li>
    <li>python39Packages.entrance</li>
    <li>python39Packages.entrance-with-router-features</li>
    <li>python39Packages.json-logging</li>
    <li>python39Packages.sanic</li>
    <li>python39Packages.sanic-auth</li>
    <li>python39Packages.sanic-routing</li>
    <li>python39Packages.sanic-testing</li>
  </ul>
</details>

**Comment by dotlambda** [comment]
Backport for Python 3.10 support?

**Comment by mweinelt** [comment]
Sounds good to me.

File: pkgs/development/python-modules/sanic/default.nix [hunk 1]
```diff
@@ -95,49 +84,27 @@ buildPythonPackage rec {
...
-    # noisy_exceptions sometimes missing from sanic stdout
-    "test_noisy_exceptions"
-  ] ++ lib.optionals (stdenv.hostPlatform.system == "aarch64-linux") [
-    # test fail on aarch64
-    "test_tls_wrong_options"
-    "test_cookie_expires"
-    "test_gunicorn_worker"
-    "test_gunicorn_worker_no_logs"
-    "test_gunicorn_worker_with_logs"
+    # OSError: foo
```
Comment (dotlambda) [inline pkgs/development/python-modules/sanic/default.nix:90]: foo?

Comment (mweinelt) [reply pkgs/development/python-modules/sanic/default.nix:90]: That's what it says.

**Review by dotlambda — APPROVED** [review]

**Pull request merged**